getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,170 @@
+
+val storage: Storage = StorageOptions.getDefaultInstance().service
+val bucket = storage.get(System.getenv("GOOGLE_STORAGE_BUCKET"))
+
+const val RESOURCES_PATH = "resources/"
+const val EMOJIS_PATH = RESOURCES_PATH + "emojis/"
+
+enum class Emoji {
+    JOY, ANGER, SURPRISE, SORROW, NONE
+}
+
+val emojiPic = mapOf(
+    Emoji.JOY to "joy.png",
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+// Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (likelihood in emotionsLikelihood) { // In this order: VERY_LIKELY, LIKELY, POSSIBLE
+        for (emotion in emotions) { // In this order: JOY, ANGER, SURPRISE, SORROW
+            if (emotion.value == likelihood) return emotion.key // Returns emotion corresponding to likelihood
+        }
+    }
+    return Emoji.NONE
+}
+
+// Downloads source image to a temp file
+fun downloadObject(objectName: String) {
+    val blob = bucket.get(objectName)
+    val localFilePath = Paths.get("/tmp/$objectName")
+    val writeTo = PrintStream(FileOutputStream(localFilePath.toFile()))
+    writeTo.write(blob.getContent())
+    writeTo.close()
+}
+
+class GcsObject(val bucketName: String, val blobName: String)
+class EmojifyResponse(val original: GcsObject, val emojified: GcsObject, val emojifiedUrl: String)
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        var publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
+
+        // Setting up image annotation request
+        val requests = ArrayList<AnnotateImageRequest>()
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://${bucket.name}/$objectName").build() // Fetching our source image
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        vision.close()
+
+        for (resp in response.responsesList) {
```

Comment:
Since we know we're only submitting one request and we should get one response, how about we just pop the first one off this list?

Reply:
Even though currently we are only annotating image for a single request (FACE_DETECTION), this sample can evolve in the future to perform other requests like LABEL_DETECTION, etc. I think iterating over a responseList makes the code more evolvable.

Reply:
the response of `emojify` method returns value for a single image. this method is designed to handle 1 image.

looping here for multiple response doesn't make sense. it expects 1 result, so:
1. it should check and validate that there is 1 item in the list, throw an exception if not the case.
2. if there is 1 and only 1 item, then operate on that item